src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -239,444 +249,455 @@ void EnFz_Destroy(Actor* thisx, PlayState* play) {
+    this->mainTimer = 2 * 20;
+    this->actionFunc = EnFz_SkatingAimFreeze;
 }
 
-void func_80933444(EnFz* this, PlayState* play) {
-    func_80933014(this);
-    if (this->unk_BCA == 0) {
-        func_80933480(this, play);
+void EnFz_SkatingAimFreeze(EnFz* this, PlayState* play) {
+    EnFz_SetYawTowardsPlayer(this);
+    if (this->mainTimer == 0) {
+        EnFz_SetupSkatingFreeze(this, play);
     }
 }
 
-void func_80933480(EnFz* this, PlayState* play) {
-    this->unk_BD6 = 1;
-    this->unk_BCA = 80;
-    this->actionFunc = func_809334B8;
-    func_80932784(this, play);
+void EnFz_SetupSkatingFreeze(EnFz* this, PlayState* play) {
+    this->state = FZ_STATE_FULLSIZE;
+    this->mainTimer = 4 * 20;
+    this->actionFunc = EnFz_SkatingFreeze;
+    EnFz_UpdateTargetPos(this, play);
 }
 
-void func_809334B8(EnFz* this, PlayState* play) {
-    Vec3f sp64;
-    Vec3f sp58;
-    Vec3f sp4C;
-    Vec3f sp40;
-    u8 sp3F;
-    s16 sp3C;
+void EnFz_SkatingFreeze(EnFz* this, PlayState* play) {
+    Vec3f baseVelocity;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
 
-    if (this->unk_BCA == 0) {
-        func_809330D4(this);
+    if (this->mainTimer == 0) {
+        EnFz_SetupDisappear(this);
         return;
     }
 
-    if (this->unk_BCA > 10) {
-        sp3F = 0;
-        sp3C = 150;
+    if (this->mainTimer > 10) {
+        u8 damaging = false;
+        s16 primAlpha = 150;
         Actor_PlaySfx_Flagged(&this->actor, NA_SE_EN_FREEZAD_BREATH - SFX_FLAG);
-        if ((this->unk_BCA - 10) < 16) {
-            sp3C = (this->unk_BCA * 10) - 100;
+        if ((this->mainTimer - 10) < 16) {
+            primAlpha = (this->mainTimer * 10) - 100;
         }
 
-        sp40.x = sp40.z = 0.0f;
-        sp40.y = 0.6f;
+        accel.x = accel.z = 0.0f;
+        accel.y = 0.6f;
 
-        sp58.x = this->actor.world.pos.x;
-        sp58.y = this->actor.world.pos.y + 20.0f;
-        sp58.z = this->actor.world.pos.z;
+        pos.x = this->actor.world.pos.x;
+        pos.y = this->actor.world.pos.y + 20.0f;
+        pos.z = this->actor.world.pos.z;
 
         Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_NEW);
 
-        sp64.x = 0.0f;
-        sp64.y = -2.0f;
-        sp64.z = ((ENFZ_GET_F(&this->actor) == ENFZ_F_1)   ? 10.0f
-                  : (ENFZ_GET_F(&this->actor) == ENFZ_F_2) ? 20.0f
-                                                           : 0.0f) +
-                 20;
+        baseVelocity.x = 0.0f;
+        baseVelocity.y = -2.0f;
+        baseVelocity.z = ((ENFZ_GET_POWER(&this->actor) == FZ_POWER_1)   ? 10.0f
+                          : (ENFZ_GET_POWER(&this->actor) == FZ_POWER_2) ? 20.0f
+                                                                         : 0.0f) +
+                         20;
 
-        Matrix_MultVec3f(&sp64, &sp4C);
+        Matrix_MultVec3f(&baseVelocity, &velocity);
 
-        if ((this->unk_BCA & 7) == 0) {
-            sp3F = 1;
+        if ((this->mainTimer & 0x7) == 0) {
```

Comment:
`% 8` doesn't work here?

Reply:
Not sure why that slipped my mind, but yes it works